File: clients/ios/TmuxRemote/Models/DeviceBookmark.swift
import Foundation

struct DeviceBookmark: Codable, Identifiable {
    var id: String { deviceId }
    var productId: String
    var deviceId: String
    var fingerprint: String
    var sct: String
    var name: String
    var agentName: String?
    var lastSession: String?
    var lastConnected: Date?

    /// Display name: user override (if set) takes priority, then agent name, then device ID.
    var displayName: String {
        if name != deviceId { return name }
        return agentName ?? deviceId
    }

    enum CodingKeys: String, CodingKey {
        case productId = "product_id"
        case deviceId = "device_id"
        case fingerprint = "device_fingerprint"
        case sct
        case name
        case agentName = "agent_name"
        case lastSession = "last_session"
        case lastConnected = "last_connected"
    }
}

File: clients/android/app/src/main/java/com/nabto/tmuxremote/models/DeviceBookmark.kt
package com.nabto.tmuxremote.models

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
data class DeviceBookmark(
    @SerialName("product_id") val productId: String,
    @SerialName("device_id") val deviceId: String,
    @SerialName("device_fingerprint") val fingerprint: String,
    val sct: String,
    val name: String,
    @SerialName("last_session") var lastSession: String? = null,
    @SerialName("last_connected") var lastConnected: Long? = null
) {
    val id: String get() = deviceId
}
